javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+
+While there are many differences, which are detailed in the chart below, in my opinion, the two major difference between `require` and `import` is first that the `require` one will automatically scan node_modules to find modules whereas because it comes from ES6, `import` will not. Second, `import` won't support node_modules since it is based on ES6, but `require` will. Effectively, `import` must specify the path of the module.
+
+#### Can we use import in node?
+
+Although you will not be seeing this a lot, it is possible if you:
+
+1. Add the `--experimental-modules` flag when running Node
+
+2. Use the `.mjs` extension or set `"type": "module"` in your `package.json`.
+
+Altghough these, like I said, are not seen often, it is worthy to know.
+
+#### Which one to use? And When?
+
+With `import` you can use named imports, import namespace, it's asynchronous... However, it's not easy to use with Node. Nonetheless, although there doesn't seem to be any performance benefit at the moment, it is still advisable to think about using `import` over `require`, since it’s about to be native in JS should later on perform better than `require`.
+
+If you want to build a full stack app and would prefer to use the same syntax for the whole application, then using `imports` is a no-brainer. However, if your app uses `__dirname` for relative file paths with the `fs` module, or if you need to import JSON files, then you would better consider using `require`.
+
+
+#### Chart of the Differences
+
+For your convienience, here is a detailed chart of the differences between `require` and `import`.
+
+|REQUIRE|IMPORT|
+|--|--|
+|Can be called at any time and place in the program.|Always run in the beginning of the file.|
+|You can directly run the code with the require statement.|If you are running Node 13 and above, to run a program containing import statement you have to use the `--experimental-modules` flag.|
+|If you want to use the require module, you have to save file with ‘.js’ extension.|If you are running Node 13 and above, if you want to use import module, you have to save file with ‘.mjs’ extension.|
+|Can leave out a .js extension when importing a local module|Cannot leave the extension|
+|Loading is synchronous, which means that if you have multiple `require`, they are loaded and processed one by one.|Import is asynchronous.|
+|Loading is synchronous, which means that if you have got multiple requires, they’re loaded and processed one by one.| Import is asynchronous|
+|To include multiple functions, you first export the functions in an object, then import them using `require`|To include multiple functions, simply import them between curly braces.|
+
+
+### Guided Practice
+
+In order to be able to follow this guided practice, make sure you are using Node.js 13 and above. To install Node, simply go to [https://nodejs.org/en/download/](https://nodejs.org/en/download/) and choose the correct installation for your system.
```

Comment:
Please remove this setup. They already have node setup and those aren't the instructions we'd suggest.

Reply:
no problem
[c89273b](https://github.com/Techtonica/curriculum/pull/1414/commits/c89273b844a10c1f67a1d0886a773f18a1c327e8)